### DISCUSSION TITLE
JS Hooks - send Mail to is not working

### DISCUSSION THREAD

**Discussion by jonathan-martz** [pocketbase/pocketbase#5787] [Q&A] [answered]
https://pocketbase.io/docs/js-sending-emails/

Iam using the "Send custom email" js Hook to send an email after an Order is created.

```
onRecordAfterCreateRequest((e) => {
    const message = new MailerMessage({
        from: {
            address: $app.settings().meta.senderAddress,
            name:    $app.settings().meta.senderName,
        },
        to:      [{address: e.record.email()}],
        subject: "YOUR_SUBJECT...",
        html:    "YOUR_HTML_BODY...",
        // bcc, cc and custom headers are also supported...
    })

    $app.newMailClient().send(message)
}, "orders")
```

but the Line is not working. Can i access somehow the @request.auth.email ?
```
to:      [{address: e.record.email()}],
```

Orders have a field called customer which is a relation to a user, so its an Id.

**Comment by ganigeorgiev** [comment, marked as answer]
The linked example in the docs is for a user registration and it is working correctly.

In your case it seems that you want to send an email after  creating an "orders" record, but from your description it is not clear to whom you want to send it - to the authenticated user (could be also admin) or to the mentioned "customer" `relation` record.

Assuming that you want the latter, then you can retrieve the related customer like using one of the methods shown in [Fetch single records](https://pocketbase.io/docs/js-records/#fetch-single-record), for example:

```js
const customer = $app.dao().findRecordById("customers", e.record.get("customer"))

// then you should be able to access "customer.email()" if customer is an auth record...
```

If you want access to the current authenticated user then you can access the current request context via `e.httpContext` and use one of the options shown in [Retrieving the current auth state](https://pocketbase.io/docs/js-routing/#retrieving-the-current-auth-state) (`c` in the example is the request context, aka. `e.httpContext`).

If you need further help, try to export your collections configuration as json (Admin UI > Settings > Export collections) and elaborate more on your use case to understand the bigger picture and the desired behavior, but again claiming that it doesn't work is wrong.
